src/theme-provider.js

```diff
@@ -16,6 +19,16 @@ class ThemeProvider extends Component {
   // create theme, by merging with outer theme, if present
   getTheme(passedTheme) {
     const theme = passedTheme || this.props.theme
+    if (isFunction(theme)) {
+      const mergedTheme = theme(this.outerTheme)
+      if (!isPlainObject(mergedTheme)) {
+        throw new Error(
```

Comment:
Yeah @goodmind - that indeed [seems to be the case](https://github.com/facebook/prop-types/blob/0445b8b52edf5e2812f3c6c58922078872b2262e/factoryWithTypeCheckers.js#L166)